spec/classes/puppet_agent_install_spec.rb

```diff
@@ -21,14 +32,16 @@
     end
 
     context "on a Windows family OS" do
-      let :facts do {
-        :osfamily => 'windows',
-        :concat_basedir => 'C:\Temp',
-        :puppetversion => Puppet.version,
-      } end
+      let :facts do
+        {
+          :osfamily => 'windows',
+          :concat_basedir => 'C:\Temp',
+          :puppetversion => Puppet.version,
+        }.merge(additional_facts)
+      end
 
       it 'should define provider as chocolatey' do
-        should contain_package('puppet').with_provider('chocolatey')
+        should contain_package(client_package).with_provider('chocolatey')
```

Comment:
If the provider is switched to MSI it can be > 4.0. I am working to test it with that provider now... will hopefully have a result in the next 2 hours.